qemu/tests/queues_number_test.py

```diff
@@ -0,0 +1,213 @@
+import logging
+import re
+import os
+
+from virttest import error_context
+from virttest import data_dir
+from virttest import utils_net
+from virttest import utils_test
+from virttest import utils_misc
+from virttest import utils_netperf
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    MULTI_QUEUE chang queues number test
+    1) Boot up VM, and login guest
+    2) Enable the queues in guest
+    3) Run netperf_and_ping test
+    4) Change queues number repeatedly during netperf_and_ping stress testing
+    5) Reboot VM
+    6) Repeat above 1-4 steps
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    def change_queues_number(ifname, q_number, queues_status=None):
+        """
+        change queues number
+        """
+        if not queues_status:
+            queues_status = get_queues_status(ifname)
+        mq_set_cmd = "ethtool -L %s combined %d" % (ifname, q_number)
+        output = session_serial.cmd_output_safe(mq_set_cmd)
+        cur_queues_status = get_queues_status(ifname)
+
+        err_msg = ""
+        expect_q_number = q_number
+        if q_number != queues_status[1] and q_number <= queues_status[0]:
+            if (cur_queues_status[1] != q_number
+                    or cur_queues_status[0] != queues_status[0]):
+                err_msg = "Param is valid, but change queues failed, "
+        elif cur_queues_status != queues_status:
+            if q_number != queues_status[1]:
+                err_msg = "Param is invalid, "
+            err_msg += "Current queues value is not expected, "
+            expect_q_number = queues_status[1]
```

Comment:
Above if block is still confusing to me, but I'm fine if others think it's ok. 
@PaulYuuu Would you please help review, thanks.

Reply:
In fact, code is copied from `mq_change_qnum.py`, although it seems difficult to understand. Fortunately, there are just error messages.
An example:
```
# ethtool -l enP2p1s0f0
Channel parameters for enP2p1s0f0:
Pre-set maximums:
RX:		0
TX:		0
Other:		1
Combined:	128
Current hardware settings:
RX:		0
TX:		0
Other:		1
Combined:	119
```

Reply:
Okay. Thanks for the info. ACK.